File: core/src/mindustry/client/antigrief/ConfigRequest.java
package mindustry.client.antigrief;

import arc.util.*;
import mindustry.gen.*;
import mindustry.world.*;

import static mindustry.Vars.*;

public class ConfigRequest {
    public final int x, y;
    public Object value;
    public boolean isRotate;

    public ConfigRequest(int x, int y, Object value, boolean isRotate) {
        this.x = x;
        this.y = y;
        this.value = value;
        this.isRotate = isRotate;
    }

    public ConfigRequest(int x, int y, Object value) {
        this(x, y, value, false);
    }

    public void run() {
        Log.info("world:" + world + "   tile:" + world.tile(x, y));
        if (world != null) {
            Tile tile = world.tile(x, y);

            if (tile == null) return;

            if (isRotate) Call.rotateBlock(player, tile.build, (boolean) value);
            else Call.tileConfig(player, tile.build, value);
        }
    }
}

File: core/src/mindustry/client/antigrief/ConfigRequest.kt
package mindustry.client.antigrief

import mindustry.*
import mindustry.gen.*

open class ConfigRequest @JvmOverloads constructor(@JvmField val x: Int, @JvmField val y: Int, var value: Any?, var isRotate: Boolean = false) : Runnable {
    @JvmOverloads constructor(build: Building, value: Any?, isRotate: Boolean = false): this(build.tileX(), build.tileY(), value, isRotate)

    override fun run() {
        val tile = Vars.world?.tile(x, y) ?: return
        if (isRotate) Call.rotateBlock(Vars.player, tile.build, value as Boolean)
        else Call.tileConfig(Vars.player, tile.build, value)
    }
}
